tools/sz_repo_cli/lib/src/commands/src/deploy_ios_command.dart

```diff
@@ -0,0 +1,304 @@
+    try {
+      final flavor = argResults![flavorOptionName] as String;
+      final stage = argResults![releaseStageOptionName] as String;
+      await runProcessSucessfullyOrThrow(
+        'fvm',
+        [
+          'dart',
+          'run',
+          'sz_repo_cli',
+          'build',
+          'ios',
+          '--flavor',
+          flavor,
+          '--stage',
+          stage,
+          '--build-number',
+          '$buildNumber',
+        ],
+        workingDirectory: _repo.sharezoneCiCdTool.path,
+      );
+    } catch (e) {
+      throw Exception('Failed to build iOS app: $e');
+    }
+  }
+
+  Future<void> _publish() async {
+    final whatsNew = argResults![whatsNewOptionName] as String?;
+    final issuerId = argResults![issuerIdOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_ISSUER_ID'];
+    final keyIdentifier = argResults![keyIdOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_KEY_IDENTIFIER'];
+    final privateKey = argResults![privateKeyOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_PRIVATE_KEY'];
+
+    final track = _getAppleTrack();
+    await runProcessSucessfullyOrThrow(
+      'app-store-connect',
+      [
+        'publish',
+        '--path',
+        'build/ios/ipa/*.ipa',
+        '--release-type',
+        // The app version will be automatically released right after it has
+        // been approved by App Review.
+        'AFTER_APPROVAL',
+        if (whatsNew != null) ...[
+          '--whats-new',
+          whatsNew,
+        ],
+        if (track is AppStoreTrack) ...[
+          '--app-store',
+        ],
+        if (track is TestFlightTrack) ...[
+          '--beta-group',
+          track.groupName,
+          '--testflight',
+        ],
+        if (issuerId != null) ...[
+          '--issuer-id',
+          issuerId,
+        ],
+        if (keyIdentifier != null) ...[
+          '--key-id',
+          keyIdentifier,
+        ],
+        if (privateKey != null) ...[
+          '--private-key',
+          privateKey,
+        ],
+      ],
+      workingDirectory: _repo.sharezoneFlutterApp.location.path,
+    );
+  }
+
+  AppleTrack _getAppleTrack() {
+    final stage = argResults![releaseStageOptionName] as String;
+    final track = _iosStageToTracks[stage];
+    if (track == null) {
+      throw Exception('Unknown track for stage: $stage');
+    }
+    return track;
+  }
+}
+
+/// A track to publish the app for Apple.
+abstract class AppleTrack {
+  const AppleTrack();
+}
+
+/// The track for the App Store.
+///
+/// https://appstoreconnect.apple.com/apps/1434868489/appstore
+class AppStoreTrack extends AppleTrack {
+  const AppStoreTrack();
+}
+
+/// The track for TestFlight.
+///
+/// https://appstoreconnect.apple.com/apps/1434868489/testflight
+class TestFlightTrack extends AppleTrack {
+  /// The name of the TestFlight group.
+  final String groupName;
+
+  const TestFlightTrack(this.groupName);
+}
```

Comment:
This will be reused for macOS. Not sure where the file should be located. In `lib/src/common/src`?

Reply:
Yes, sounds good 👍